panic: runtime error: invalid memory address or nil pointer dereference

**What were you trying to accomplish?**
The `eksctl` panics when trying to deploy a cluster with the `ssm.enableSsm` enabled.

**What happened?**
```
2021-08-28 03:08:10 [!]  SSM is now enabled by default; `ssh.enableSSM` is deprecated and will be removed in a future release
2021-08-28 03:08:10 [ℹ]  eksctl version 0.63.0
2021-08-28 03:08:10 [ℹ]  using region us-east-1
2021-08-28 03:08:10 [ℹ]  subnets for us-east-1a - public:192.168.0.0/19 private:192.168.96.0/19
2021-08-28 03:08:10 [ℹ]  subnets for us-east-1b - public:192.168.32.0/19 private:192.168.128.0/19
2021-08-28 03:08:10 [ℹ]  subnets for us-east-1c - public:192.168.64.0/19 private:192.168.160.0/19
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x27097ae]

goroutine 1 [running]:
github.com/weaveworks/eksctl/pkg/ctl/create.doCreateCluster(0xc000622790, 0xc00007dc00, 0xc0005bdb80, 0x0, 0x0)
        github.com/weaveworks/eksctl/pkg/ctl/create/cluster.go:198 +0x60e
github.com/weaveworks/eksctl/pkg/ctl/create.createClusterCmd.func1(0xc000622790, 0xc00007dc00, 0xc0005bdb80, 0xc0005bdb80, 0x3889240)
        github.com/weaveworks/eksctl/pkg/ctl/create/cluster.go:37 +0x3f
github.com/weaveworks/eksctl/pkg/ctl/create.createClusterCmdWithRunFunc.func1(0xc00055d400, 0xc00007da00, 0x0, 0x2, 0x0, 0x0)
        github.com/weaveworks/eksctl/pkg/ctl/create/cluster.go:56 +0x12c
github.com/spf13/cobra.(*Command).execute(0xc00055d400, 0xc00007d9e0, 0x2, 0x2, 0xc00055d400, 0xc00007d9e0)
        github.com/spf13/cobra@v1.2.1/command.go:856 +0x472
github.com/spf13/cobra.(*Command).ExecuteC(0xc00055c000, 0xc0001f3760, 0x1, 0xc0008dff70)
        github.com/spf13/cobra@v1.2.1/command.go:974 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.2.1/command.go:902
main.main()
        github.com/weaveworks/eksctl/cmd/eksctl/main.go:77 +0x41b
```

**How to reproduce it?**

 1. Create a cluster with `eksctl create cluster -f` with the following configuration:

```yaml
kind: ClusterConfig
apiVersion: eksctl.io/v1alpha5
metadata:
  name: sandbox
  version: "1.21"
  region: us-east-1
availabilityZones:
  - us-east-1a
  - us-east-1b
  - us-east-1c
addons:
  - name: vpc-cni
    version: latest
  - name: kube-proxy
    version: latest
  - name: coredns
    version: latest
managedNodeGroups:
  - name: sandbox-amd64-workers
    instanceType: m5.large
    desiredCapacity: 3
    minSize: 2
    maxSize: 5
    privateNetworking: true
    ssh:
      enableSsm: true
```

**Anything else we need to know?**

```
$ uname -a
Linux ip-172-31-4-200.ec2.internal 4.14.243-185.433.amzn2.x86_64 #1 SMP Mon Aug 9 05:55:52 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```

**Versions**
```
$ eksctl info
eksctl version: 0.63.0
kubectl version: v1.21.2-13+d2965f0db10712
OS: linux
```

--- a/pkg/apis/eksctl.io/v1alpha5/types.go
+++ b/pkg/apis/eksctl.io/v1alpha5/types.go
@@ -1638,13 +1638,3 @@ type UnsupportedFeatureError struct {
 func (u *UnsupportedFeatureError) Error() string {
 	return fmt.Sprintf("%s: %v", u.Message, u.Err)
 }
-
-// DeprecationError represents a deprecation error.
-// +k8s:deepcopy-gen=false
-type DeprecationError struct {
-	Message string
-}
-
-func (d *DeprecationError) Error() string {
-	return d.Message
-}

--- a/pkg/apis/eksctl.io/v1alpha5/validation.go
+++ b/pkg/apis/eksctl.io/v1alpha5/validation.go
@@ -9,6 +9,7 @@ import (
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/kris-nova/logger"
 	"github.com/pkg/errors"
 
 	corev1 "k8s.io/api/core/v1"
@@ -274,9 +275,7 @@ func validateNodeGroupBase(ng *NodeGroupBase, path string) error {
 			if !*enableSSM {
 				return errors.New("SSM agent is now built into EKS AMIs and cannot be disabled")
 			}
-			return &DeprecationError{
-				Message: "SSM is now enabled by default; `ssh.enableSSM` is deprecated and will be removed in a future release",
-			}
+			logger.Warning("SSM is now enabled by default; `ssh.enableSSM` is deprecated and will be removed in a future release")
 		}
 	}
 

--- a/pkg/ctl/cmdutils/cmd.go
+++ b/pkg/ctl/cmdutils/cmd.go
@@ -39,18 +39,10 @@ func (c *Cmd) NewCtl() (*eks.ClusterProvider, error) {
 		logger.Warning("ignoring validation error: %s", err.Error())
 	}
 
-	checkDeprecationError := func(err error) error {
-		if _, ok := err.(*api.DeprecationError); ok {
-			logger.Warning(err.Error())
-			return nil
-		}
-		return err
-	}
-
 	for i, ng := range c.ClusterConfig.NodeGroups {
 		if err := api.ValidateNodeGroup(i, ng); err != nil {
 			if c.Validate {
-				return nil, checkDeprecationError(err)
+				return nil, err
 			}
 			logger.Warning("ignoring validation error: %s", err.Error())
 		}
@@ -62,7 +54,7 @@ func (c *Cmd) NewCtl() (*eks.ClusterProvider, error) {
 	for i, ng := range c.ClusterConfig.ManagedNodeGroups {
 		api.SetManagedNodeGroupDefaults(ng, c.ClusterConfig.Metadata)
 		if err := api.ValidateManagedNodeGroup(ng, i); err != nil {
-			return nil, checkDeprecationError(err)
+			return nil, err
 		}
 	}
 

